amethyst_core/src/transform/components/transform.rs

```diff
@@ -643,144 +639,59 @@ impl From<Vector3<f64>> for Transform {
-
-                let isometry = Isometry3::from_parts(
-                    Translation3::new(translation[0], translation[1], translation[2]),
-                    Unit::new_normalize(Quaternion::new(
-                        rotation[3],
-                        rotation[0],
-                        rotation[1],
-                        rotation[2],
-                    )),
-                );
-                let scale = Vector3::new(scale[0], scale[1], scale[2]);
-
-                Ok(Transform {
-                    isometry,
-                    scale,
-                    ..Default::default()
-                })
-            }
-        }
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "Transform", default)]
+struct TransformValues {
+    translation: [Float; 3],
+    rotation: [Float; 4],
+    scale: [Float; 3],
+}
 
-        const FIELDS: &[&str] = &["translation", "rotation", "scale"];
-        deserializer.deserialize_struct("Transform", FIELDS, TransformVisitor::default())
+impl Default for TransformValues {
+    /// The default transform does nothing when used to transform an entity.
+    fn default() -> Self {
+        TransformValues {
+            translation: [Float::zero(); 3],
+            rotation: [Float::zero(), Float::zero(), Float::zero(), Float::one()],
+            scale: [Float::one(); 3],
+        }
     }
 }
 
-impl Serialize for Transform {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        #[derive(Serialize)]
-        struct TransformValues {
-            translation: [Float; 3],
-            rotation: [Float; 4],
-            scale: [Float; 3],
+impl From<TransformValues> for Transform {
+    fn from(transform_values: TransformValues) -> Self {
+        let TransformValues {
+            translation,
+            rotation,
+            scale,
+        } = transform_values;
+
+        let isometry = Isometry3::from_parts(
+            Translation3::new(translation[0], translation[1], translation[2]),
+            Unit::new_normalize(Quaternion::new(
+                rotation[3],
+                rotation[0],
+                rotation[1],
+                rotation[2],
+            )),
+        );
+        let scale = Vector3::new(scale[0], scale[1], scale[2]);
+
+        Transform {
+            isometry,
+            scale,
+            ..Default::default()
         }
+    }
+}
 
-        let pos: [Float; 3] = self.isometry.translation.vector.into();
-        let rot: [Float; 4] = self.isometry.rotation.as_ref().coords.into();
-        let scale: [Float; 3] = self.scale.into();
-
-        Serialize::serialize(
-            &TransformValues {
-                translation: [pos[0], pos[1], pos[2]],
-                rotation: [rot[0], rot[1], rot[2], rot[3]],
-                scale: [scale[0], scale[1], scale[2]],
-            },
-            serializer,
-        )
+impl Into<TransformValues> for Transform {
```

Comment:
a little pedantic, but if the following is implemented instead, we get the `Into<TransformValues> for Transform` for free from std's blanket impl.

```rust
impl From<Transform> for TransformValues {
     fn from(transform: Transform) -> TransformValues {
        TransformValues {
            translation: transform.isometry.translation.vector.into(),
            rotation: transform.isometry.rotation.as_ref().coords.into(),
            scale: transform.scale.into(),
        }
    }
}
```

The main advantage is, when `Into` is implemented, then this happens:

```rust
let transform_values = transform.into(); // ok!
let transform_values = TransformValues::from(transform); // fails
```

whereas if you implement `From`, then both work. Normally you'd only implement `Into` if the target type is not under your control.